internal/ingress/controller/controller.go

```diff
@@ -527,6 +527,12 @@ func (n *NGINXController) getBackendServers(ingresses []*ingress.Ingress) ([]*in
 				addLoc := true
 				for _, loc := range server.Locations {
 					if loc.Path == nginxPath {
+						// Same paths but different types are allowed
+						// (same type means overlap in the path definition)
+						if *loc.PathType != *path.PathType {
+							break
+						}
```

Comment:
Given there are 3 ingresses (ordered by creation timestamp):

1. example.com, /example, prefix, backend1
2. example.com, /example, exact, backend2
3. example.com, /example, prefix, backend3

I'd expect after applying these ingresses in the same order to get a prefix location pointing to backend3 and exact location pointingg to backend2. However reading the code I think the controller will create a prefix location pointing to backend1, not backend3 because we are breaking out of loop as soon as pathtype does not match.

In other words, why are you not doing `if loc.Path == nginxPath && *loc.PathType != *path.PathType {` instead?

Reply:
> However reading the code I think the controller will create a prefix location pointing to backend1,

This is the right behavior. Similar to what we have now without PathType